index.d.ts

```diff
@@ -0,0 +1,25 @@
+export interface runJxa {
+    /**
+     * Returns a Promise for the value returned from input.
+     *
+     * @param {string | function} input - .
+     * If a function, it's stringified and passed to JXA. It should be atomic, meaning it doesn't access anything outside its body.
+     * If a string, you can access the specified arguments with args array. Use the arguments parameter rather than template interpolation so you don't have to do escaping.
+     * @param {!Array[any]} args - Arguments to pass to the JXA context. Items should be serializable (JSON.stringify'able).
+     * @returns Returns a Promise for the value returned from input.
+     */
+    (input: string | ((...args:Array<string>) => void), args?: Array<any>):Promise<any>;
```

Comment:
I would use [function overloads](https://www.typescriptlang.org/docs/handbook/functions.html#overloads). Then you can define 2 different signatures for the same method.

```ts
type JSONStringifyable = object | number | string | boolean;

(input: string, args?: JSONStringifyable[]): Promise<JSONStringifyable>;
<T, U>(input: (...args: U[]) => T, args: U[]): Promise<T>;
```

Something like that, needs some more finetuning perhaps but it's a good start.

Reply:
Use shorthand syntax for array.